From 32bdd94391cbd88c6792beac63fc762d6fb2ade4 Mon Sep 17 00:00:00 2001
From: amodra <amodra@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Wed, 10 Sep 2014 05:02:28 +0000
Subject: [PATCH 237/326] 	PR debug/60655 	* dwarf2out.c
 (mem_loc_descriptor <PLUS>): Return NULL if addend 	can't be output.

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_8-branch@215103 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/ChangeLog   | 6 ++++++
 gcc/dwarf2out.c | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/gcc/ChangeLog b/gcc/ChangeLog
index ecd14a4..9a34c8e 100644
--- a/gcc/ChangeLog
+++ b/gcc/ChangeLog
@@ -1,3 +1,9 @@
+2014-09-10  Alan Modra  <amodra@gmail.com>
+
+	PR debug/60655
+	* dwarf2out.c (mem_loc_descriptor <PLUS>): Return NULL if addend
+	can't be output.
+
 2014-09-09  Richard Biener  <rguenther@suse.de>
 
 	Backport from mainline
diff --git a/gcc/dwarf2out.c b/gcc/dwarf2out.c
index 77771c0..f387e61 100644
--- a/gcc/dwarf2out.c
+++ b/gcc/dwarf2out.c
@@ -12234,7 +12234,7 @@ mem_loc_descriptor (rtx rtl, enum machine_mode mode,
 	      op1 = mem_loc_descriptor (XEXP (rtl, 1), mode, mem_mode,
 					VAR_INIT_STATUS_INITIALIZED);
 	      if (op1 == 0)
-		break;
+		return NULL;
 	      add_loc_descr (&mem_loc_result, op1);
 	      add_loc_descr (&mem_loc_result,
 			     new_loc_descr (DW_OP_plus, 0, 0));
-- 
2.0.3

